fertiscan/db/queries/label/__init__.py

```diff
@@ -124,23 +134,19 @@ def get_label_information_json(cursor, label_info_id) -> dict:
     Returns:
     - dict: The label information in json format.
     """
-    try:
-        query = """
-            SELECT get_label_info_json(%s);
-            """
-        cursor.execute(query, (str(label_info_id),))
-        label_info = cursor.fetchone()
-        if label_info is None or label_info[0] is None:
-            raise LabelInformationNotFoundError(
-                "Error: could not get the label information: " + str(label_info_id)
-            )
-        return label_info[0]
-    except LabelInformationNotFoundError as e:
-        raise e
-    except Exception as e:
-        raise e
+    query = """
+        SELECT get_label_info_json(%s);
+        """
+    cursor.execute(query, (str(label_info_id),))
+    label_info = cursor.fetchone()
+    if label_info is None or label_info[0] is None:
+        raise LabelInformationNotFoundError(
+            "Error: could not get the label information: " + str(label_info_id)
+        )
+    return label_info[0]
 
 
+@handle_query_errors(LabelInformationQueryError)
```

Comment:
and here we use `LabelInformationQueryError` but raise `LabelInformationNotFoundError`. Why is it in one case a query error and the other a retrival error. Also what is the difference between a retrival error and NotFoundError. I like how the handle_querry_error() works, but I dont get how you use it in this case. (2/2)

Reply:
Label dimension is not really a LabelInformation. Using LabelInformationRetrievalError did not seem appropriate, hence, I use LabelInformationQueryError which is the "default" exception for LabelInformations. But if this does not seem natural, please suggest an alternative.

Note that I now raise LabelInformationQueryError as well when no data is returned.

Reply:
So from my understanding:
If when retrieving a label and no data is return (result = None )we get `LabelInformationNotFoundError` which raise `LabelInformationRetrievalError` and we also trigger `LabelInformationQueryError`  ?
If there is a random error during the retrieval of a label or label_dimension we get `LabelInformationQueryError`
If when retrieving a label_dimension and no data is return (result = None ) we only get `LabelInformationQueryError`? 

I understand that a label_dimension is not the same as a label_information. (Its an abstract concept representing the data of the real thing) I think this issue has a goal of giving more insight on what is happening. **I'd give it a retrieval or not found type of error to help the user debug**(maybe even a new exception).

 Side note, the function is mostly used in testing and if a label_information doesn't exist, it's dimension should not exists.

Reply:
I created new errors for label dimension. Seems neater.

Note: I've updated the description of the PR to explain what's going on.